toml.abnf

```diff
@@ -84,9 +84,10 @@ ml-basic-string = ml-basic-string-delim ml-basic-body ml-basic-string-delim
 
 ml-basic-string-delim = 3quotation-mark
 
-ml-basic-body = *( ml-basic-char / newline / ( escape ws newline ) )
-ml-basic-char = ml-basic-unescaped / escaped
+ml-basic-body = *( ml-basic-char / wschar / newline / ml-basic-escaped-nl )
+ml-basic-char = ( ml-basic-unescaped / escaped )
 ml-basic-unescaped = wschar / %x21-5B / %x5D-7E / non-ascii
+ml-basic-escaped-nl = escape newline *( wschar / newline )
```

Comment:
This seems to remove the whitespace between the escape and the newline?

Reply:
@ehuss You're right. The rule should start with `escape ws newline` rather than `escape newline`, since the written spec clearly says that whitespace is allowed here: "When the last non-whitespace character on a line is a `\` ...".

Reply:
And `escape ws newline` (that was here at the beginning) along with the extra `*( wschar / newline )` added seems to be fitting that sentence.  Both under the new token matching all input to skip.

The changeset is more minimal now.

Reply:
My apologies, @josuah, for pushing you the wrong way. I misread the rule in the spec and gave you bad advice. I'm sorry.

Reply:
@eksortso I am to blame as much for not checking the clear and unambiguous sentence "last non-whitespace character on a line is a `\`".

Should we rephrase it like so it is more foolproof (1) ?

```diff
-When the last non-whitespace character on a line is a \,
+If a \ is followed by (optional) whitespaces and then a newline,
```

The original sentence is still fine if you stick to decomposing its accurate meaning and do not read what you want to see.

(1): https://1.bp.blogspot.com/-HucRsU8S-OM/XMxkwBN4uoI/AAAAAAAACdE/IdF0VaUqXsI8xt_7OjAurjlp_47pAyEoACPcBGAYYCw/s1600/MultilineString.png